### DISCUSSION TITLE
Can I run the app layer in Crazyflie simulation?

### DISCUSSION THREAD

**Discussion by amitx64** [bitcraze/discussions#1443] [Q&A]
Hi everyone,

I'm working on a Crazyflie simulation and I'm wondering if it's possible to run the `app layer` within the simulation environment.

Specifically, I'm interested in understanding how to simulate the behavior of apps that typically run on the Crazyflie hardware. Has anyone had experience with this, or are there any guidelines or tools available to achieve this?

Thanks in advance for your help!

**Comment by knmcguire** [comment]
Hi! Currently we don't have support for python bindings for app layers: There have been some questions about it somewhere in the discussions, including this one: https://github.com/orgs/bitcraze/discussions/1000, but we haven't implemented that functionality yet.

For the rest, you could look at these blogposts for some more SITL versions of the Crazyflie, but those are external projects that we don't maintain, so you'll need to look through these yourself:
- https://www.bitcraze.io/2024/04/crazysim-a-software-in-the-loop-simulator-for-the-crazyflie/
- https://www.bitcraze.io/2024/01/simulation-sim_cf2-for-multi-agent-algorithm-development/
